metashare/api/models.py

```diff
@@ -502,6 +513,30 @@ def add_commits(self, commits, sender):
     def add_ms_git_sha(self, sha):
         self.ms_commits.append(sha)
 
+    def queue_submit_review(self, *, user, data):
+        from .jobs import submit_review_job
+
+        self.currently_submitting_review = True
+        self.save()
+        self.notify_changed()
+        submit_review_job.delay(user=user, task=self, data=data)
+
+    def finalize_submit_review(
+        self, timestamp, err=None, status=None, delete_org=False
+    ):
+        self.currently_submitting_review = False
+        if err:
+            self.save()
+            self.notify_error(err, "TASK_SUBMIT_REVIEW_FAILED")
+        else:
+            self.review_submitted_at = timestamp
+            self.review_status = status
+            self.review_valid = True
+            self.save()
+            self.notify_changed("TASK_SUBMIT_REVIEW")
```

Comment:
@wlonk It's difficult for me to test locally (together you and I could test on staging), but I'm worried we have a bug in how we're using WS events that we should discuss...

Now that pretty much every user is subscribing to every model they know about (repo, project, task, org), and we're showing a success/failure on a lot of the WS events (e.g. "pr created", "org created" etc), I'm worried that _all_ users will be seeing those messages instead of just the user that triggered the job. For many events I've been able to filter on the front end (e.g. "only show this to the user who actually owns the org"), but for other events I don't have an easy way to know what user triggered the action (i.e. any user can create a task PR -- who do we show the success message to?).

I think ideally we'd have a pattern for two types of events fired from the back-end (e.g. "TASK_UPDATED" and "TASK_PR_CREATED"), or perhaps even simpler we just always include the `user.id` for the user who created the job in the WS event payload. Thoughts?

Reply:
@wlonk These are the places where I'm worried we're currently displaying user-facing messages to users who did not trigger the action that resulted in a WS event:

- submit project for review, success (`PROJECT_CREATE_PR`)
- submit project for review, error (`PROJECT_CREATE_PR_FAILED`)
- re-sync repo users, error (`REPOSITORY_UPDATE_ERROR`)
- submit task for review, success (`TASK_CREATE_PR`)
- submit task for review, error (`TASK_CREATE_PR_FAILED`)
- submit review on task, success (`TASK_SUBMIT_REVIEW`)
- submit review on task, error (`TASK_SUBMIT_REVIEW_FAILED`)

I'm open to your thoughts or suggestions. Brainstorming options, in no particular order:

- Add fields to each of those models to track who "owns" a given action (e.g. `project.pr_owner`, `task.pr_owner`, `task.review_owner` -- but what do we do about resync-repo-users?)
- Add one-off WS channels that users subscribe to when triggering an action (outside the existing "subscribe to all events for this model" pattern)
- Add triggering-user-id to WS event payloads (requires passing through from view/request, to job, back to model method that fires WS message)
- ...others?

Reply:
@jgerigmeyer I feel like the scope of these changes are about the same (read: may require changes across a lot of different areas of the code). The option to add an owner to (above) actions sounds most appealing, and maybe `resync-repo-users` could be special and display a toast for just the currently logged-in user.

Reply:
Moved to a dedicated card: https://trello.com/c/MdAoQYx8